h2/src/main/org/h2/util/DriverPropertyInfoSet.java

```diff
@@ -0,0 +1,79 @@
+            newProp("OPTIMIZE_EVALUATABLE_SUBQUERIES", true, "Database setting OPTIMIZE_EVALUATABLE_SUBQUERIES (default: true). \nOptimize subqueries that are not dependent on the outer query."),
+            newProp("OPTIMIZE_INSERT_FROM_SELECT", true, "Database setting OPTIMIZE_INSERT_FROM_SELECT (default: true). \nInsert into table from query directly bypassing temporary disk storage. \nThis also applies to create table as select."),
+            newProp("OPTIMIZE_IN_LIST", true, "Database setting OPTIMIZE_IN_LIST (default: true). \nOptimize IN(...) and IN(SELECT ...) comparisons. \nThis includes optimization for SELECT, DELETE, and UPDATE."),
+            newProp("OPTIMIZE_IN_SELECT", true, "Database setting OPTIMIZE_IN_SELECT (default: true). \nOptimize IN(SELECT ...) comparisons. \nThis includes optimization for SELECT, DELETE, and UPDATE."),
+            newProp("OPTIMIZE_OR", true, "Database setting OPTIMIZE_OR (default: true). \nConvert (C=? OR C=?) to (C IN(?, ?))."),
+            newProp("OPTIMIZE_SIMPLE_SINGLE_ROW_SUBQUERIES", true, "Database setting OPTIMIZE_SIMPLE_SINGLE_ROW_SUBQUERIES (default: true). \nOptimize expressions of the form (SELECT A) to A."),
+            newProp("OPTIMIZE_TWO_EQUALS", true, "Database setting OPTIMIZE_TWO_EQUALS (default: true). \nOptimize expressions of the form A=B AND B=1. \nIn this case, AND A=1 is added so an index on A can be used."),
+            newProp("QUERY_CACHE_SIZE", 8, "Database setting QUERY_CACHE_SIZE (default: 8). \nThe size of the query cache, in number of cached statements. \nEach session has it's own cache with the given size. \nThe cache is only used if the SQL statement and all parameters match. \nOnly the last returned result per query is cached. \nThe following statement types are cached: SELECT statements are cached (excluding UNION and FOR UPDATE statements), CALL if it returns a single value, DELETE, INSERT, MERGE, UPDATE, and transactional statements such as COMMIT. \nThis works for both statements and prepared statement."),
+            newProp("RECOMPILE_ALWAYS", false, "Database setting RECOMPILE_ALWAYS (default: false). \nAlways recompile prepared statements."),
+            newProp("REUSE_SPACE", true, "Database setting REUSE_SPACE (default: true). \nIf disabled, all changes are appended to the database file, and existing content is never overwritten. \nThis setting has no effect if the database is already open."),
+            newProp("SHARE_LINKED_CONNECTIONS", true, "Database setting SHARE_LINKED_CONNECTIONS (default: true). \nLinked connections should be shared, that means connections to the same database should be used for all linked tables that connect to the same database."),
+            newProp("ZERO_BASED_ENUMS", false, "Database setting ZERO_BASED_ENUMS (default: false). \nIf set, ENUM ordinal values are 0-based."),
```

Comment:
- There is no `MV_STORE` setting.
- `ZERO_BASED_ENUMS` is going to be removed, don't list it here too.
- I think it's a bad idea to hard-code defaults here, it will be better to grab actual defaults from `DbSettings` or other sources.
- Many settings here (`OPTIMIZE_*` and possibly others) are just kill switches. If some bug in optimization will be found, such switch can be used as a temporary workaround until this bug will be fixed. I think it is a bad idea to return them here, because they can be misused by accident.
- `DEFAULT_TABLE_ENGINE` shouldn't be listed too, it exists for exotic use cases in third-party solutions.

I think it will be better to list only useful settings here, including some settings from other sources, take a look on `ConnectionInfo`, it contains names of various other settings, including popular ones. @grandinj, what do you think?

You also need to break long lines, our code formatting rules don't allow lines longer than 120 characters.